tools_recognition:
  ros__parameters:

    threshold: 0.8

    start_on_init: false
      
    max_sizes:
        - [0.05, 0.05, 0.05]

    model_file: yolov8n.pt

    services:
    
      object_recognition:
        start: /micky_vision/recognition_start
        stop: /micky_vision/recognition_stop

    subscribers:
      
      slop: 0.2
      qos_profile: 10

      image_rgb: /micky_vision/camera/color/image_raw
      camera_info: /micky_vision/camera/color/camera_info
      image_depth: /micky_vision/camera/aligned_depth_to_color/image_raw

    publishers:

      object_recognition:
        topic: /micky_vision/tools_recognition
        qos_profile: 10

      debug:
        topic: /micky_vision/tools_classes
        qos_profile: 10